src/rulesets/packages/SpecialFunctions.jl

```diff
@@ -67,6 +67,11 @@
     (SpecialFunctions.digamma(a) - SpecialFunctions.digamma(a + b),
      SpecialFunctions.digamma(b) - SpecialFunctions.digamma(a + b),)
 )
+if isdefined(SpecialFunctions, :expint)
+    @scalar_rule(
+        SpecialFunctions.expint(ν, z), (NaN, -SpecialFunctions.expint(ν - 1, z)),
```

Comment:
In principle `expint` is differentiable wrt `ν`, right? I wonder which is better here, `DoesNotExist`, `NaN`, or something else? Because we don't mean that the derivative is not a number or even that it doesn't exist. We mean that we don't support that derivative yet. @oxinabox what do you think?

Reply:
I think neither `NaN` nor `DoesNotExist` serve our purpose here.

I really think it would be best that we wait on the implementation of `meijerg` function in `SpecialFunctions` before we implement the `scalar_rule` for `expint`. Though ofcourse I'm open to change my mind, if we see good reason for that.

Reply:
> We mean that we don't support that derivative yet.

Then we can't write the rule.
Need to write the whole rule or no rule at all.
Writing a rule disabled the AD from doing its thing, so we need to give an answer

>  I really think it would be best that we wait on the implementation of meijerg function in SpecialFunctions before we implement the scalar_rule for expint. Though ofcourse I'm open to change my mind, if we see good reason for that.

Sounds like a plan